### PR TITLE
Speed up conda build `conda_build` Circle test by using `mamba` solver via `boa` (and use it for Github Actions test too)

### LINKED ISSUE #1242: `conda_build` test taking in excess of an hour and failing often on Circle
Conda is capricious again - I'll have a look now what's going on

### PR DESCRIPTION
**Pull request by valeriupredoi**
## Description

Speeding up `conda_build` test using `boa` ([boa on conda](https://anaconda.org/conda-forge/boa) and [boa documentation](https://github.com/mamba-org/boa) - boa is part of the mamba project) and its extension to use `mamba` solver during build instead of the `conda` solver. Speedup of a few factors; tested with local build and on [Github Actions](https://github.com/ESMValGroup/ESMValCore/pull/1243/checks?check_run_id=3201971674)

Closes #1242 

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by valeriupredoi** [comment]
I am getting royally confused about what the actual process on the Circle machine is - `conda build` takes about 15-17min on my laptop, the env setup with Python 3.9 that I am using in the PR took 50min yesterday and now it's past 1h25min and it's still not done, and should take much less given how fast is locally on my laptop - am I missing something here? What you think @bouweandela

File: .circleci/config.yml [hunk 1]
```diff
@@ -181,13 +181,16 @@ jobs:
             set -x
             # Install prerequisites
             mkdir /logs
-            # conda update -y conda > /logs/conda_base.txt 2>&1
+            conda update -y conda > /logs/conda_base.txt 2>&1
+            conda create --name build 'python=3.9'
+            set +x; conda activate build; set -x
             conda install -y conda-build conda-verify ripgrep >> /logs/conda_base.txt 2>&1
+            conda install -y -c conda-forge mamba
```
Comment (jvegasbsc) [inline .circleci/config.yml:188]:
```suggestion
            conda create --name build -c conda-forge 'python=3.9' conda-build conda-verify ripgrep mamba >> /logs/conda_base.txt 2>&1
            set +x; conda activate build; set -x
```

**Comment by valeriupredoi** [comment]
cheers @jvegasbsc - more compact indeed, but I don't think that'll do any good with the time it takes to run though, I am baffled as to why it takes so long on the CI machine and so fast on me laptop :confused:

**Comment by jvegasbsc** [comment]
> but I don't think that'll do any good with the time it takes to run though, I am baffled as to why it takes so long on the CI machine and so fast on me laptop 😕

It should help, as it only tries to resolve the environment once... If that improvement it's meaningful or not, that's another story

**Comment by valeriupredoi** [comment]
OK let's just merge this one, it consistently does better than what we have now, I am still scratching my head why so slow though - retested on me laptop again and even followed exactly the same CI steps from a brand new Miniconmda install as it does it on the CI machine and got 19-25min - the CI machine is dam slow, that's what it is :rofl:

**Comment by valeriupredoi** [comment]
can we get a yay or a nay here, folks, pls :beer:

**Comment by valeriupredoi** [comment]
OK `conda_build` is actually fixed and sped up considerably ie from an hour on the CI machine to 9 minutes; this is experimental though, I am using `conda mambabuild` with `boa` which is the conda package builder with `mamba` as solver instead of `conda`, it should be fine since all tests pass fine, but I'll err on the side of caution for now and will have to compare the dependency trees (tomorrow)

**Comment by zklaus** [comment]
Let us know when this is really, really ready, ok? :laughing:

**Comment by valeriupredoi** [comment]
on it now, bud, I'll set the really REALLY ready label after lunch, provided stuff's the right way up :grin:

**Comment by valeriupredoi** [comment]
OK guys, I looked at the deps and even retro-installed the package from the local build, all looks fine and works fine, I see no reason why we shouldn't adopt this - makes our lives much better, I declare it ready-READY :grin: Now on to figuring out the ESMValTool build :man_facepalming:

**Comment by zklaus** [comment]
LGTM. Once the checklist is done and we have proper labels and stuff, I'll approve.

**Comment by valeriupredoi** [comment]
on it as we shpeak :+1:

**Comment by valeriupredoi** [comment]
this PR has put its make up on and is ready to be merged, at your leisure @zklaus or @bouweandela or @jvegasbsc :beer:

**Review by zklaus — APPROVED** [review]

**Pull request merged**